### PR TITLE
Fix struct destructors

### PR DESCRIPTION
**Pull request by scottinet**
# Description

* Add missing destructor for the `date_result` struct
* Fix how the user_rights struct is handled:
  * Convert the user_rights underlying results from a NULL-terminated array to a size-delimited one
  * Rename the user_rights result property from "result" to "rights", and add a "rights_length" properties, making it on par with every other structs
  * Fix the destructor which was not freeing the memory correctly

### CODE REVIEW

File: cgo/kuzzle/destructors.go [hunk 1]
```diff
@@ -495,6 +504,15 @@ func kuzzle_free_int_result(st *C.int_result) {
 	}
 }
 
+//export kuzzle_free_date_result
```
Comment (jenow) [inline cgo/kuzzle/destructors.go:507]: Why did you add this, I don't see any usage of it? Is it a part of another PR?

**Review by jenow — CHANGES_REQUESTED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

File: cgo/kuzzle/destructors.go (hunk 1 above)
Comment (scottinet) [reply cgo/kuzzle/destructors.go:507]: Yes, it is used in the fixed version of `server:now`: https://github.com/kuzzleio/sdk-cpp/pull/44/files#diff-ad60d65b34e16a3319296bb5d683acd6R79

**Review by jenow — APPROVED** [review]
